plugin/mlsdbonlinelocator.cpp

```diff
@@ -79,7 +79,7 @@ void MlsdbOnlineLocator::defaultVoiceModemChanged(const QString &modem)
 
 bool MlsdbOnlineLocator::haveFieldData(const QList<MlsdbProvider::CellPositioningData> &cells)
 {
-    return !(cells.isEmpty() && m_wifiServices.isEmpty() && (!m_simManager || !m_simManager->isValid()));
+    return !((cells.isEmpty() || !m_simManager || !m_simManager->isValid()) && m_wifiServices.isEmpty());
```

Comment:
I would write this condition like this to make it clearer:

`if (!cells.isEmpty() || (m_simManager && m_simManager->isValid()) || !m_wifiServices.isEmpty())`

I.e., if there is celltowers in cells list or if you are connected to cellular network or there are wifi services in the service list.

Reply:
The purpose was such, that not send a request if a list of cell towers and WiFi networks list is empty.  But in this case the query will still be executed

Reply:
"if you are connected to a cellular network you have info at least one network right"  
then return !((cells.isEmpty() || !m_simManager || !m_simManager->isValid()) && m_wifiServices.isEmpty())  --->  return !((0 || 0 || 0) && 1);   ---> return 1   ---> 
if (!haveFieldData(cells))   ---> if(0)  In this way, the request is sent to the server. 
We use this information!